commit 7690346248fd8d3460cc07f018827ff55b85d19d
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Sat Aug 24 14:00:34 2013 +0100

    sna: Don't copy fbcon if hosted
    
    When hosted, we have no idea what the right initial contents should be
    as we are independent of the CRTC and so the existing fb is irrelevant.
    Plus, not actually being master will cause the attempt to read back the
    bound framebuffer to fail...
    
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/src/sna/sna_driver.c b/src/sna/sna_driver.c
index 83a5bba..e6c424a 100644
--- a/src/sna/sna_driver.c
+++ b/src/sna/sna_driver.c
@@ -259,7 +259,7 @@ static Bool sna_create_screen_resources(ScreenPtr screen)
 	screen->SetScreenPixmap(sna->front);
 
 	/* Only preserve the fbcon, not any subsequent server regens */
-	if (serverGeneration == 1)
+	if (serverGeneration == 1 && (sna->flags & SNA_IS_HOSTED) == 0)
 		sna_copy_fbcon(sna);
 
 	if (!sna_become_master(sna)) {
